### ISSUE TITLE
textgenrnn gets stuck in loops

### ISSUE THREAD

**Issue by paulhoule** [minimaxir/textgenrnn#19]
This behavior is visible at In[3] in the notebook

https://github.com/minimaxir/textgenrnn/blob/master/docs/textgenrnn-demo.ipynb

```
Trump confirms a complete shot of the state of the state of the state of the state of the state of the state of the season to the state of the state of my parents for the state of the man who was a good day.
```

this phenomenon can be avoided by raising the temperature but this results in clearly bad results.

**Comment by minimaxir** [comment]
That's just the case with the untrained set trying to compress a bunch of information into a relatively small model. This will not happen on trained datasets.

There could be better examples in the docs but getting the pretrained model to a good size has tradeoffs.

**Comment by paulhoule** [comment]
No,  it is not that.

I have the same troubles when I run on 5 million chars of somewhat curated training data.  (Whole paragraphs of RFC text)

I am happy to tolerate longer training times,  but when I see the loss go to 0.9 I say WTF because I have never seen a model I've trained go below 1 for real and I've seen models w/ a cross-entropy loss go to 1.15 that do not repeat themselves like that.

Anyway,  I do epoch after epoch under these conditions and it does not seem to get better over time.

I was going to attach a recipe for generating my training data for this ticket but then I noticed you were having the same looping behavior on the documentation example.

**Comment by paulhoule** [comment]
I tried an entirely new model with

```
rnn.config['rnn_size']=1024
rnn.config["max_size"]=1024
```

I ran 20 epochs with a 1.5 million character training set and a claimed loss of 0.615

Results are better than before but the looping is still awful.  At T=0.2

```
o  If the server receives a request to the server to send a reply cache and the client is allowed to use the client to set a request to the server to access the server to the server to the server to indicate that the client sends a LAYOUTRETURN operation to the client to return the same server own
```

At T=0.5 I am now getting results that aren't quite so embarassing...

```
Servers associated with the server, the server MUST NOT return NFS4ERR_WRONGSEC.  The server should change the client that the file is used to restrict the server to interpret a regular file, the server MUST return NFS4ERR_NOFILEHANDLE.
```

**Comment by minimaxir** [comment]
I would not recommend using `rnn_size=1024` unless you have an order-of-magnitude more data. Even then that's pushing it. Try doing 128x6 or 256x3.

`max_size` is not a parameter; `max_length` is. Assuming you meant `max_length`, a good rule of thumb is to set `max_length` to half the median of the document size. If you have it set too big, the model will train poorly as well.

**Comment by minimaxir** [comment]
After more testing on various datasets, I recommend doing an architecture of `max_length=40`, `rnn_size=128`, `run_layers=6`, or if your documents are really long, `80/256/3`. Only if those approaches don't work should you get more fancy with architecture, but that's not a fault with the code itself.
